spec/patch.spec.js

```diff
@@ -23,10 +23,12 @@ describe('keyword $patch', function() {
           "source": {
             "type": "object",
             "properties": { "p": { "type": "string" } },
-            "additionalProperties": false
+            "additionalProperties": false,
+            "required": [ "p" ]
           },
           "with": [
-            { "op": "add", "path": "/properties/q", "value": { "type": "number" } }
+            { "op": "add", "path": "/properties/q", "value": { "type": "number" } },
+            { "op": "add", "path": "/required/-", "value": "q" }
```

Comment:
if you want to add "required" keyword and its change with "patch", then it has to be added with "merge" as well (it can replace the whole array) and also tested in "test_validate.js" (object without "q" should fail); as is it is not tested at all.

Reply:
thank you for the review.  Changes submitted